packages/iris-grid/src/IrisGridPartitionSelector.tsx

```diff
@@ -88,35 +105,58 @@ class IrisGridPartitionSelector<T> extends Component<
     this.sendFetchAll();
   }
 
-  handlePartitionChange(event: React.ChangeEvent<HTMLInputElement>): void {
+  handlePartitionChange(
+    index: number,
+    event: React.ChangeEvent<HTMLInputElement>
+  ): void {
     log.debug2('handlePartitionChange');
 
-    const { column } = this.props;
+    const { columns } = this.props;
+    const { partitions, partitionTables } = this.state;
     const { value: partition } = event.target;
 
+    const newPartitions = [...partitions];
+    newPartitions[index] =
+      TableUtils.isCharType(columns[index].type) && partition.length > 0
+        ? partition.charCodeAt(0).toString()
+        : partition;
+    if (partitionTables) {
+      this.updatePartitionFilters(newPartitions, partitionTables);
+    }
+
     this.setState({
-      partition:
-        TableUtils.isCharType(column.type) && partition.length > 0
-          ? partition.charCodeAt(0).toString()
-          : partition,
+      partitions: newPartitions,
     });
 
     this.debounceUpdate();
   }
 
-  handlePartitionSelect(partition: string): void {
-    if (this.searchMenu) {
-      this.searchMenu.closeMenu();
+  handlePartitionSelect(index: number, partition: string): void {
+    const { columns } = this.props;
+    const { partitions, partitionTables } = this.state;
+    const selectedMenu = this.searchMenu[index];
+    if (selectedMenu) {
+      selectedMenu.closeMenu();
+    }
+
+    const newPartitions = [...partitions];
+    newPartitions[index] =
+      TableUtils.isCharType(columns[index].type) && partition.length > 0
+        ? partition.charCodeAt(0).toString()
```

Comment:
Why do we need to limit it to one char here? Why are we allowed to enter more than one char in this case at all?